packages/common-lib/src/renderers/reports/sqr601.ts

```diff
@@ -54,6 +55,35 @@ export const sqr601DocumentDefinition = (
 		return aName.localeCompare(bName);
 	}
 
+	function getHFZExpire(reqs: CadetPromotionStatus): string {
+		let dateVal = '';
+		if (reqs.CurrentCadetAchv.CadetAchvID < 5 && Array.isArray(reqs.HFZRecords)) {
+			if (reqs.HFZRecords.length === 0) {
+				return '';
+			} else {
+				dateVal = reqs.HFZRecords[0].DateTaken.substr(0, 10);
+				return new Date(
+					new Date(dateVal).getTime() + 60 * 60 * 24 * 182 * 1000,
+				).toLocaleDateString('en-US');
+			}
+			// return 'Phase I';
+		} else if (Array.isArray(reqs.HFZRecords)) {
+			if (reqs.HFZRecords.filter(rec => rec.IsPassed === true).length > 0) {
+				dateVal = reqs.HFZRecords.filter(rec => rec.IsPassed === true)[0].DateTaken.substr(
+					0,
+					10,
+				);
+				return new Date(
+					new Date(dateVal).getTime() + 60 * 60 * 24 * 182 * 1000,
+				).toLocaleDateString('en-US');
+			} else {
+				return '';
+			}
+		} else {
+			return '-'; // shouldn't ever get here...
+		}
+	}
```

Comment:
We know that it is an array, and there's also functions like `find` or `some` that would be a better fit, and also the `=== true` checks, so maybe:

```suggestion
	function getHFZExpire(reqs: CadetPromotionStatus): string {
		if (reqs.CurrentCadetAchv.CadetAchvID < 5) {
			if (reqs.HFZRecords.length === 0) {
				return '';
			} else {
				const dateVal = reqs.HFZRecords[0].DateTaken.substr(0, 10);
				return new Date(
					new Date(dateVal).getTime() + 60 * 60 * 24 * 182 * 1000,
				).toLocaleDateString('en-US');
			}
			// return 'Phase I';
		} else {
			if (reqs.HFZRecords.some(rec => rec.IsPassed)) {
				const dateVal = reqs.HFZRecords.find(rec => rec.IsPassed).DateTaken.substr(
					0,
					10,
				);
				return new Date(
					new Date(dateVal).getTime() + 60 * 60 * 24 * 182 * 1000,
				).toLocaleDateString('en-US');
			} else {
				return '';
			}
		}
	}
```

I'm also a bit confused as to the logic for only using the first record for Phase I + C/SSgt; C/SSgt being part of Phase II should be lumped with the other group, right? Also, why the difference in the first place?

Reply:
The way I wrote it was the only way I could get it to work.  When I attempted to access .length directly, it would fail on undefined object.  If you would like to rewrite it, you are welcome to do so, or you could approve the code that I wrote that works.  Oops.  That should have been < 4, not < 5.

Reply:
The difference is because Phase I cadets don't have to pass the PT test, they just have to attempt it to "establish a baseline"

Reply:
What about if they have multiple baselines? Does CAPWATCH data allow this?

Reply:
We should be importing all CAPWATCH HFZ records, then we can sift, sort, and collate as required.  The way I wrote it, the latest HFZ record will be used to establish timing for Phase I cadets' 6 month window and the latest passing HFZ record will be used to establish timing for all other cadets' 6 month window.

Reply:
The way you wrote it, the first HFZ record is used. What is the first HFZ record according to CAPWATCH data?

Reply:
we need to import all because if a cadet passes, then fails and we only import the last CAPWATCH record, then we won't be able to refer to the last passing score.

Reply:
Disregard, I checked the server code and it is limiting to one and sorting based off of HFZID

Reply:
The records are sorted by HFZID DESC, so the first record is always the newest record.

Reply:
Wait, why is it an array type then, instead of a MaybeObj?

Reply:
I tried using a MaybeObj but didn't get it to work.  I got it to work as an array.

Reply:
Good enough for now, I guess